### PR TITLE
EtherSocial Network (ESN) support added

### PR DESCRIPTION
**Pull request by ghost**
- EtherSocial Network (ESN) support added.
  - Homepage: https://ethersocial.org/
  - Explorer: https://ethersocial.net/ 
  - Node: https://api.esn.gonspool.com
  - ESN status: http://escstats.gonsmine.com/
  - BIP44 / Chainid 31102 ( https://github.com/satoshilabs/slips/pull/277 )
- TREZOR support for EtherSocial Network (ESN) added.

### CODE REVIEW

File: app/scripts/nodes.js [hunk 1]
```diff
@@ -20,6 +20,7 @@ nodes.nodeTypes = {
     EGEM: "EGEM",
     CLO: "CLO",
     EAST: "EAST",
+    ESN: "ESN",
```
Comment (ghost) [inline app/scripts/nodes.js]: Why are you willing to put ESN on the middle position?

**Comment by akx20000a** [comment]
And please add color code for ESN selection [example](https://github.com/kvhnuke/etherwallet/pull/1876/files#diff-fdb517236c6450fac305b50634838a19R598)

File: app/scripts/nodes.js [hunk 2]
```diff
@@ -312,6 +313,19 @@ nodes.nodeList = {
         'service': 'easthub.io',
         'lib': new nodes.customNode('https://node.easthub.io', '')
     },
+    'esn': {
+        'name': 'ESN',
+        'blockExplorerTX': 'https://ethersocial.net/tx/[[txHash]]',
+        'blockExplorerAddr': 'https://ethersocial.net/addr/[[address]]',
+        'type': nodes.nodeTypes.ESN,
+        'eip155': true,
```
Comment (akx20000a) [inline app/scripts/nodes.js]: EIP-155 is not enabled for ESN network yet, we can wait for it 😄

File: app/scripts/nodes.js [hunk 3]
```diff
@@ -313,6 +314,19 @@ nodes.nodeList = {
         'service': 'easthub.io',
         'lib': new nodes.customNode('https://node.easthub.io', '')
     },
+    'esn': {
```
Comment (ghost) [inline app/scripts/nodes.js]: @kimmyeonghun @hackmod Please move ESN to the bottom of the nodes.js also

**Review by ghost — CHANGES_REQUESTED** [review]

File: app/scripts/nodes.js (hunk 3 above)
Comment (hackmod) [reply app/scripts/nodes.js]: @akx20000a // done.

> kimmyeonghun added some commits 29 days ago

as you can see, this works already done and prepared ready to merge but, as you know.. hardfork schedule was postponed internally. sorry for inconvince

**Review by ghost — APPROVED** [review]

**Comment by hackmod** [comment]
oops! missing `break;` added and rebased.

**Comment by hackmod** [comment]
rebased again (6/15)

**Comment by hackmod** [comment]
@gamalielhere // any problem with this PR? please let me know

**Comment by hackmod** [comment]
@eosclassicteam // thank you for your concern

This is my test on some api servers with the following rpc test  in Korea.
`{"jsonrpc":"2.0","method":"eth_getBlockByNumber","params":["0x0",true],"id":1}`

network | API server | response time
---------|-----------|-----------
 ETH | https://mainnet.infura.io/mew | 1.215 sec
 ETH | https://mew.giveth.io | 6.759 sec
 Ropsten testnet | https://ropsten.infura.io/mew | 1.164sec
 EXP | https://node.expanse.tech/ | *failed* (cert error)
ELLA | https://jsonrpc.ellaism.org | 1.660sec
CLO | https://clo-geth.0xinfra.com/ | 2.494sec
ESN | https://api.esn.gonspool.com | 1.225sec

I think It is enough

**Comment by eosclassicteam** [comment]
@hackmod Ok how much delay for https://node.eos-classic.io in korea can you test it?

**Comment by hackmod** [comment]
rebased

**Comment by akx20000a** [comment]
Ethersocial Network is a scam!!! DO NOT MERGE THIS PR!!!!!!

**Comment by hackmod** [comment]
image

@gamalielhere // do not feed the troll
he is well known serial hard forker in Korea.

Ethersocial Network was born at the biggest miner community in Korea https://www.ddengle.com/

See also https://github.com/MyCryptoHQ/MyCrypto/pull/1881

**Comment by akx20000a** [comment]
default


LOL You guys said you will list ESN on Bit-Z several months ago.... now there is nothing 😆

**Comment by hackmod** [comment]
Please do not use fake or modified images.

this is the original post written in Korean.
https://www.ddengle.com/esn/7837146

-----
update 7/6

ESN will be listed on Bit-z next week
- Deposit open: 2018.07.09
- Trading open: 2018.07.11
ANN @ bit-z - https://support.bit-z.pro/hc/en-us/articles/360004599953

See also https://support.bit-z.pro/hc/en-us/articles/360004597793-ESN-Ethersocial-

**Comment by akx20000a** [comment]
@hackmod LOL only time will & would tell. There is no such coin that takes like 2 months for listing on a single exchange unless you guys are lying.... ddengle P2P thread is the worst case ddengle will end up for buying & selling their coin themselves 😃

**Comment by akx20000a** [comment]
They started their mainnet **7 months** ago and there is **no exchange** with ESN support!!!

Also their fund is backed by "Youbit", which is a scam exchange also!!

**Comment by akx20000a** [comment]
@magnalucus @kimmyeonghun Why don't you guys compensate your presale ethereum funds to Youbit hack victims??

**Comment by hackmod** [comment]
thank you for your concern.
your trolling is recorded at github.
image

**Comment by akx20000a** [comment]
아 제가 딴지거는것은 인과응보 라는겁니다^^ https://github.com/kvhnuke/etherwallet/pull/1610

4달전에 이 자리에서 깽판치신건 기억 못하시고 여전히 줏대없이 경거망동하시네요 박완X씨^^

그리고 저희도 ESN 식으로 말하면 7월 신규 상장 및 호재 있습니다 덤핑하고 튀었다는 뇌내망상 하지 마시기 바랍니다

**Comment by hackmod** [comment]
image

Thank you for your concern

**Comment by akx20000a** [comment]
@wbobeirne Please close this PR, they are trying to impersonate Ethereum Social [ ETSC ] and they are telling so much rumors out of nowhere. 

Proof they are trying to attempt illegal impersonation & spreading bad rumors of Ethereum Social)
a
b
c
d
e

Ethereum Social is the only coin that runs with normal blockchain development & professional development company located on Korea.

We run Ethereum Classic public node api endpoint service for faster connection, and some Callisto blockchain based projects with Ethereum Commonwealth.

EtherSocial Network is only composed of some volunteers with small payment and hired marketers for fishing miners on korea.

We don't need to support forked, scam coin for their localized scam for selling their premined coins,
It will only result as confusing ETSC and ESN for investors and users using the coin.

**Comment by wkpark** [comment]
hello. I am real @hackmod (you can checkout my real profile at github)
we do not wanted to make much noise for this PR at your nice project, sorry for inconvenience.

https://www.ddengle.com/ is the biggest miner/cryptocurrency community in Korea (since 2013),
the EtherSocail Network (ESN) was incubated at this community several years ago by the community manager named 쌍둥아빠(@magnalucus)

but to be clear, the following is updated current status of ESN and ETSC

EtherSocial Network (ESN) | Ethereum Social (ETSC)
------------------------|----------
https://ethersocialnetwork.com | http://ethereumsocial.kr/ 
White paper https://github.com/ethersocial/wiki/blob/master/pages/white-paper/%5Benglish%5D-white-paper.md | FIXME (disappered)
Community support (Since 2013~ biggest and most popular blockchain/mining community in Korea) : https://www.ddengle.com |  FIXME
Devs : @h-s-o-o , @kimmyeonghun , @magnalucus (FIXME) // external developers @hackmod (me), @ComBba (sejun pool) @turbobit (gonspool) @freewinter (mobile wallet, topmining pool)  | @akx20000a , @bitminings (almost do nothing. owner of https://http://www.gazua.tv )
Source code release date (Jan 31, 2018) : https://github.com/ethersocial/go-esn (See also the attached image)  | a fork of the original ESN (Feb 9, 2018)
Network Hash rate : ~250GH/s (https://ethersocial.net/stats/hashrate ) | ~5GH/s (https://explorer.ethereumsocial.kr/stats/hashrate )
no official pool site available // about ~7. https://esc.gonsmine.com/ https://esn.topmining.co.kr/ https://comining.io https://esn.solopool.org/ | https://pool.ethereumsocial.kr (official) https://comining.io https://etsc.solopool.org/ http://etsc.altpool.pro
network state (peer explorer) : https://escstats.gonsmine.com | https://stats.ethereumsocial.kr/
Feed back site : https://www.ddengle.com/esn | http://www.gazua.tv/post/ho
Exchange : in progress (p2p price: ~3500 satoshi ) | https://stocks.exchange/trade/ETSC/BTC (price ~10 satoshi)

image

currently, the genesis hash between ESN and ETSC are different
- ETSC is a hard fork of the ESN(formally known as ESC) at block height 150,000 (fact)
- ETSC trash their former block chain with no reason(FIXME) and start mining again from the genesis with snapshot allocation of the their former blockchain. (their snapshot allocation info found at https://github.com/paritytech/parity/blob/master/ethcore/res/ethereum/social.json
- the original ESN genesis.json for parity found at https://github.com/OpenCommunityCoin/parity/blob/ethersocial/ethcore/res/ethereum/ethersocial.json
- I've made some genesis alloc checker https://github.com/wkpark/esn-genesis-check
~~~
$ node check.js
...
Total ESN = 499224895.246414219
Total ETSC = 10055699039.79956214899999998

Total 417 accounts matched / 45 accounts not matched /  11 accounts ESN only
Total 2470 accounts ETSC only
~~~

**Comment by akx20000a** [comment]
Bitcoin Platinum Project will be restarted with the support of DigiFinex exchange ( 19th on exchange list ) and Genius Hentai Developer WJ Cloud. We will start a war with those accumulated evils, ESN, Youbit, Yapizon, WK something, and Ddengle. Those guys abused the technology of ethereum and they did nothing but abusing the power of korean mining community. ALL HAIL TO MY GLORIOUS LEADER!!!

**Comment by ghost** [comment]
@gamalielhere

If you have any problems with this PR please let me know.

See also

 - https://github.com/EthereumCommonwealth/etherwallet/pull/158
 - https://github.com/MyCryptoHQ/MyCrypto/pull/1881

**Comment by kvhnuke** [comment]
We are looking into multiple other sources for verification, we'll merge it once everything checks out. Thank you!

**Comment by hackmod** [comment]
sorry for being noisy, but to be clear
about 1 month ago, this issue posted and finally, ESN will be listed on Bit-z next week
- Deposit open: 2018.07.09
- Trading open: 2018.07.11
ANN @ bit-z - https://support.bit-z.pro/hc/en-us/articles/360004599953

See also https://support.bit-z.pro/hc/en-us/articles/360004597793-ESN-Ethersocial-

**Comment by eosclassicteam** [comment]
**RED ALERT this coin dropped more than 50% this week with more than 2% of total supply, developers are dumping the coin!!! https://www.bit-z.com/exchange/esn_btc**

**Comment by hackmod** [comment]
image

thankyou for your concern.

> 2% of total supply, developers are dumping the coin!!!

show me the source.

**Comment by akx20000a** [comment]
@hackmod 아니 그냥 거래소 화면 들어가서 거래량 보면 뻔히 몇백만개씩 거래되는거 보이는데

증거 내놓으라 이 ㅈ ㅋㅋㅋㅋㅋㅋㅋㅋㅋㅋ

**Comment by akx20000a** [comment]
ESN is a SCAM COIN with only parameter fix from go-ethereum!!!

**DO NOT MERGE THIS PR~!**

**Comment by hackmod** [comment]
thank you for your concern.
image

Sorry for noisy, but to be clear, ESN is not a scam. please do not *TROLLING* this is not your house.
you can check the balance changes of the genesis alloc if you want.

-----
so, what is your source ? prove it. I guess you do not understand the meaning of volume in the chart.

image

- the mining cost of electricity charge is too high in South Korea.
  - see also https://www.elitefixtures.com/blog/post/2683/bitcoin-mining-costs-by-country/
  - > South Korea comes in as the most expensive country for mining a single coin
- in Korea, ESN miners think of the lowest price was regraded as `3000 satoshi`
- avg. volume(24h) in the airdrop period - about ~30 BTC (peak 40 BTC)
- avg. volume(24h) after airdrop event ended - about ~1 BTC (very low volume) - **dumping period**

See also
- bit-z 350,000 ESN airdrop event for trader - https://v2.bit-z.com/?activity_id=72
- bit-z.com ESN/BTC chart https://www.bit-z.com/exchange/esn_btc

**Comment by turbobit** [comment]
@akx20000a 
I do not know why you are doing this.

**Comment by doctorgu** [comment]
@akx20000a 님은 비난받기 위해서 비난하시는 것 같습니다. 억울해지고 싶어서 계속 자기 무덤을 파는 것 같아요.

**Comment by akx20000a** [comment]
아니 ㅋㅋㅋ 마이닝 수익이 여기서 왜 나와요 곤지암 이런데 가면 몇백기가씩 캐는 애들이 수두룩 빡빡한데 ㅋㅋㅋㅋ 물론 이더나 제크 캐죠 그런분들은

@hackmod @wkpark @turbobit @doctorgu 그리고 정신 좀 차리세요 아직도 ESN 의 농간에 놀아나지 마시고요, 몇십 비트 찍던 볼륨이 단 하루만에 0.X 비트로 떨어지고 매도 매수 호가 다 바닥나고 입금은 되나 출금은 안되는 이런 상황이 코드만 배낀 코인에게 있어서 이상하다고 생각되지 않습니까??

**Comment by akx20000a** [comment]
@hackmod Please don't do anything before you go into your grave old man

guess what?
This Korean old man's recent behavior...
almost all opensource wiki contributor in Korea already know about him...

**Comment by akx20000a** [comment]
@akx20000a ESN is a continuation of Youbit and Yapizon hack scam proudly brought by Hanmudang and professional poor scammers although they had their exchange site and community site skills they only earned little profit by scamming people from nothing, that's the most important fact.

dude. Please came down and relax. your future is bright and promising, except this old man ttelddak behavior.

**Comment by akx20000a** [comment]
아 이자리에 떙글분들 다 오셨으니까 공개적으로 경고하겠습니다,

저는 작년 12월 비트코인 해외시세 고점일때 공매도 하고 트위터 한줄로 암호화폐 시종 50조를 한순간에 날린 사람입니다

ESN 도 가격 2자리수 아니 1자리수 되기 싫으면 좀 적당히 나대시고요, ESN 으로부터 떙글을 돌려주시기 바랍니다

이상입니다.

**Comment by akx20000a** [comment]
aaa

정말 @hackmod 님은 적폐세력 지지하는 협력자인것 같습니다

**Comment by akx20000a** [comment]
tlqkf

**Comment by hackmod** [comment]
poor guy..
he continuously try to leak out  my real life personal information (TEL, address) on github, discord channels, and swearing in Korean...

**Comment by hackmod** [comment]
@akx2000a account is deleted now. https://github.com/akx20000a and all his comments are almost deleted.
Im not sure why this happen but I guess..
he is continuously expose my personal information and cursed me in Korean at github, so.. maybe he is banned by github.

**Comment by hackmod** [comment]
The latest Ledger blue-app-eth support EIP155 + chainId > 255 correctly.

Ledger support added, rebased now.

**Comment by kvhnuke** [comment]
Thanks @hackmod, we are waiting on #1979 once thats merged we would be happy to merge this as well

**Review by gamalielhere — APPROVED** [review]

**Comment by hackmod** [comment]
finally, @akx2000a has removed himself from the github....

https://github.com/akx20000a removed.
https://github.com/ethereumsocial removed.
https://pool.ethereumsocial.kr removed.
https://ethereumsocial.kr/ removed.

**Pull request merged**